pkg/cloudprovider/plugins/aws/aws_account_impl.go

```diff
@@ -23,6 +25,9 @@ import (
 
 // AddProviderAccount adds and initializes given account of a cloud provider.
 func (c *awsCloud) AddProviderAccount(client client.Client, account *crdv1alpha1.CloudProviderAccount) error {
+	for idx := range account.Spec.AWSConfig.Region {
```

Comment:
Modifying the retrieved CR here, seems inappropriate.
Going forward, we may consume the region field before calling this function.